processing/src/main/java/org/apache/druid/query/QueryRunner.java

```diff
@@ -21,14 +21,13 @@
 
 import org.apache.druid.guice.annotations.ExtensionPoint;
 import org.apache.druid.java.util.common.guava.Sequence;
-
-import java.util.Map;
+import org.apache.druid.query.context.ResponseContext;
 
 @ExtensionPoint
 public interface QueryRunner<T>
 {
   /**
    * Runs the given query and returns results in a time-ordered sequence.
    */
-  Sequence<T> run(QueryPlus<T> queryPlus, Map<String, Object> responseContext);
+  Sequence<T> run(QueryPlus<T> queryPlus, ResponseContext responseContext);
```

Comment:
Since nearly all of the tests and benchmarks are doing `ResponseContext.createEmtpy()`, should we add

```suggestion
  Sequence<T> run(QueryPlus<T> queryPlus, ResponseContext responseContext);
  
  @VisibleForTesting
  default Sequence<T> run(QueryPlus<T> queryPlus)
  {
    return this.run(queryPlus, ResponseContext.createEmpty());
  }
```

to reduce boilerplate?

Reply:
looks nice. will commit the suggestion

Reply:
added and refactored tests